content/en/docs/reference/using-api/api-concepts.md

```diff
@@ -340,48 +374,70 @@ Clients that receive a response in `application/vnd.kubernetes.protobuf` that do
 waiting for a signal (field value, external system, or other) produced by a
-component responsible for a finalizer later in the list, resulting in a deadlock.
-Without enforced ordering finalizers are free to order amongst themselves and
+component responsible for a finalizer later in the list, and the finalization is
+unable to proceed.
+
+Without enforced ordering, finalizers are free to order amongst themselves and
 are not vulnerable to ordering changes in the list.
+{{< /note >}}
 
-Once the last finalizer is removed, the resource is actually removed from etcd.
+After a finalizer completes its tasks, the relevant controller marks that completion
+by removing its own entry from the list of finalizers on that resource.
 
+Once the last finalizer is removed, the resource is actually removed from etcd and,
+consequently, from the API.
 
-## Single resource API
+See [Garbage Collection](/docs/concepts/workloads/controllers/garbage-collection/) for
+details about built-in finalization for cleaning up dependents.
 
-API verbs GET, CREATE, UPDATE, PATCH, DELETE and PROXY support single resources only.
+## Single- and multiple-resource API verbs {#single-resource-api}
+
+In the API, verbs Get, Create, Update, Patch, Delete and Proxy support single
+resources only.
 These verbs with single resource support have no support for submitting
 multiple resources together in an ordered or unordered list or transaction.
-Clients including kubectl will parse a list of resources and make
-single-resource API requests.
+Clients, including `kubectl`, parse a list of resources and make
+a series of single-resource API requests if needed.
 
-API verbs LIST and WATCH support getting multiple resources, and
-DELETECOLLECTION supports deleting multiple resources.
+The API verbs List and Watch support getting multiple resources;
+DeleteCollection supports deleting multiple resources.
 
-## Dry-run
+## Dry run mode {#dry-run}
 
  {{< feature-state for_k8s_version="v1.18" state="stable" >}}
 
-The modifying verbs (`POST`, `PUT`, `PATCH`, and `DELETE`) can accept requests in a _dry run_ mode. Dry run mode helps to evaluate a request through the typical request stages (admission chain, validation, merge conflicts) up until persisting objects to storage. The response body for the request is as close as possible to a non-dry-run response. The system guarantees that dry-run requests will not be persisted in storage or have any other side effects.
+In the Kubernetes API, HTTP verbs that modify data (`POST`, `PUT`, `PATCH`, and `DELETE`) can accept requests in a _dry run_ mode. Dry run mode helps to evaluate a request through the typical request stages (admission chain, validation, merge conflicts) up until persisting objects to storage. The response body for the request is as close as possible to a non-dry-run response. The system guarantees that dry-run requests will not be persisted in storage or have any other side effects.
```

Comment:
nit? "non-dry-run" ?
The query parameter is "dryRun".

Reply:
non-dry-run is idiomatic UK English; I can't guess at the US version - can you recommend a replacement?

Reply:
The double hyphen word combination looks odd to me.
"dry run" vs "dry-run"

Reply:
I've omitted this change from the latest revision.